### ISSUE TITLE
Incorrect results because KIND specifier is now dropped from constants 

### ISSUE THREAD

**Issue by clementval** [omni-compiler/omni-compiler#177]
In older version of the `F_Front` (1.1.0 or 1.1.1), the kind specifier was kept as is and the generated code was giving identical results with the original code. 

With newer version of `F_Front`, the kind specifier is often dropped or replaced and this gives wrong numerical results in the transformed code.

We have smth declared like this in a module: 
```fortran 
  integer, parameter :: sp = selected_real_kind(6)
  integer, parameter :: dp = selected_real_kind(13)
```

Then in the code, it is used in various place (declaration, assignment and computation). 

```fortran
REAL(KIND=dp) :: my_dp_var
!
var_x = 8.0_sp

res = fct1(a,b) * 10.0_sp
```

In the past, everything was kept nicely for the constant: 
```fortran
REAL(KIND=dp) :: my_dp_var
!
var_x = 8.0_sp

res = fct1(a,b) * 10.0_sp
```

But know we have this: 
```fortran
REAL(KIND=selected_real_kind(13)) :: my_dp_var
!
var_x = 8.0

res = fct1(a,b) * 10.0
```

The XcodeML/F has drop this information for constant. 
before: 
```xml
<FrealConstant type="R1ed72b0" kind="dp">0.0</FrealConstant>
```
now:
```xml
<FrealConstant type="R1ed72b0">0.0</FrealConstant>
```

I don't think it makes a problem for the declaration but it does for the constants in the assignments and binary expression. 

Is there a quick way to recover the old behaviour? or to fix it. I'll gladly look into it if you have an entry point where I can start. 

@mitsuhisa-sato @h-murai @shingo-s

**Comment by clementval** [comment]
I figure out the problem is that when the XcodeML/F is generated, the `KIND_TYPE` is a `FUNCTION_CALL` and therefore, for a constant, nothing is generated. We should keep the `IDENTIFIER` in this case I think.

**Comment by clementval** [comment]
Fix by #178
